### PR TITLE
Cherry-pick #6604 from main. IRuntimeFactory signature change

### PR DESCRIPTION
**Pull request by andre4i**
Remove `existing` property - Part 4, IRuntimeFactory (#6604) **(except for RuntimeFactoryHelper)**

See #3429

optional argument for IRuntimeFactory.instantiateRuntime to replace the existing state
RuntimeFactoryHelper is to be used by current clients of IRuntimeFactory (see RFC - remove use of the existing property across (some) layers #6486 for examples) in a future PR, after the version bump in container definitions. This class now holds the wrapper logic about how the runtime is actually instantiated.

### CODE REVIEW

**Comment by andre4i** [comment]
Missed a commit in this. Converting it to draft until fixed.

**Comment by vladsud** [comment]
@anthony-murphy - can you please glance and provide feedback?
I think we should not bring new functionality into 0.39.
definitions - maybe, but I'd rather see us developing in main (including definitions) than jumping to old branches.
Is there an option to public minor (i.e. additive) updates from main?

Note that this is only first part, future change would be breaking (existing flag being required). While we will ensure that it stays optional from runtime behavior perspective, we want it to be compile-time breaking to push people consuming new (future) versions of loader & definitions to change their behavior, and thus need to eventually make it required.
Where and how we will be able to make such change?

**Comment by anthony-murphy** [comment]
> @anthony-murphy - can you please glance and provide feedback?
> I think we should not bring new functionality into 0.39.
> definitions - maybe, but I'd rather see us developing in main (including definitions) than jumping to old branches.
> Is there an option to public minor (i.e. additive) updates from main?
> 
> Note that this is only first part, future change would be breaking (existing flag being required). While we will ensure that it stays optional from runtime behavior perspective, we want it to be compile-time breaking to push people consuming new (future) versions of loader & definitions to change their behavior, and thus need to eventually make it required.
> Where and how we will be able to make such change?

I've talking about this with Andrei, and i think the change is safe. I agree the current process is not ideal, and am working on a proposal for something better, but for now this is the best we have. #6540

**Comment by vladsud** [comment]
I'd still not bring new feature (implementation) into 0.39 unless it's truly required.
As it needs to be ported into 0.40, 0.41, 0.42, ...
And if someone consumes it in 0.39 and finds a bug, we would need to consider fixing it in 039+.

**Comment by anthony-murphy** [comment]
it doesn't need to be ported. we have not release container definition beyond 0.39.*

**Comment by vladsud** [comment]
implementation = runtime-utils changes.

File: packages/runtime/runtime-utils/src/runtimeFactoryHelper.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
```
Comment (anthony-murphy) [inline packages/runtime/runtime-utils/src/runtimeFactoryHelper.ts:2]: i wouldn't make this change in 0.39. it won't be consumed

**Review by vladsud — APPROVED** [review]

**Pull request merged**